packages/hint/src/lib/cli/analyze.ts

```diff
@@ -17,16 +19,100 @@ import askForConfirm from '../utils/misc/ask-question';
 import cutString from '../utils/misc/cut-string';
 import * as resourceLoader from '../utils/resource-loader';
 import { installPackages } from '../utils/npm';
+import * as insights from '../utils/appinsights';
 
 const each = promisify(async.each);
 const debug: debug.IDebugger = d(__filename);
+const configStoreKey: string = 'run';
 
 /*
  * ------------------------------------------------------------------------------
  * Private
  * ------------------------------------------------------------------------------
  */
 
+const printFrame = (message: string) => {
+    logger.log(boxen(message, {
+        align: 'center',
+        margin: 1,
+        padding: 1
+    }));
+};
+
+/**
+ * Prints a message asking user to accept send telemetry data.
+ */
+const showTelemetryMessage = () => {
+    const message: string = `Help us improve webhint
+by sending limited usage information
+(no personal information or URLs will be sent).
+
+To know more about what information will be sent please
+visit ${chalk.default.green('https://webhint.io/docs/user-guide/telemetry')}`;
+
+    printFrame(message);
+};
+
+/**
+ * Prints a message asking user to configure the telemetry telemetry.
+ */
+const showCITelemetryMessage = () => {
+    const message: string = `Help us improve webhint
+by sending limited usage information
+(no personal information or URLs will be sent).
+
+To know more about what information will be sent please
+visit ${chalk.default.green('https://webhint.io/docs/user-guide/telemetry')}
+
+Please configure it using
+the environment variable HINT_TRACKING to 'on' or 'off'
+or set the flag --tracking=on|off`;
+
+    printFrame(message);
+};
+
+/** Ask user if he wants to activate the telemetry or not. */
+const askForTelemetryConfirmation = async (config: Configuration) => {
+    if (insights.isConfigured()) {
+        return;
+    }
+
+    if (isCI) {
+        if (!insights.isConfigured()) {
+            showCITelemetryMessage();
+        }
+
+        return;
+    }
+
+    const alreadyRun: boolean = configStore.get(configStoreKey);
+
+    if (!alreadyRun) { /* This is the first time, don't ask anything. */
+        configStore.set(configStoreKey, true);
+
+        return;
+    }
+
+    showTelemetryMessage();
+
+    const message: string = `Do you want to opt-in?`;
+
+    debug(`Prompting telemetry permission.`);
+
+    const confirm: boolean = await askForConfirm(message);
+
+    if (confirm) {
+        insights.enable();
+
+        insights.trackEvent('FirstRun');
```

Comment:
Shouldn't we rename the event to `SecondRun`? 😛

Reply:
done! :)